### ISSUE TITLE
Unexpected results from test model when running TOM

### ISSUE THREAD

**Issue by RubReh** [minar09/cp-vton-plus#9]
There seems to be artifacts when running TOM testing. Does this suggest that the input to the TOM is incorrectly given or are you seeing the same results as below? The input is fixed as reported here: #8 FYI.

![bild](https://user-images.githubusercontent.com/38507268/89031916-bf083380-d333-11ea-981f-2cea80096fd8.png)

![bild](https://user-images.githubusercontent.com/38507268/89031926-c3cce780-d333-11ea-902e-41f19f951a2e.png)

**Comment by minar09** [comment]
No, this is not expected as well. Here, looks like two people got merged. Same issue from listdir? Can you do the same input debugging here as well?

**Comment by minar09** [comment]
Corrected segmentation will look like this: 
![image](https://user-images.githubusercontent.com/3684134/89032433-9dc53880-d36f-11ea-847a-0373ff90a4e7.png)

**Comment by minar09** [comment]
And Final TOM output will be like this: 
![image](https://user-images.githubusercontent.com/3684134/89032452-aae22780-d36f-11ea-803e-5d0f0c1bbb16.png)

**Comment by RubReh** [comment]
Yes, the segmentation is correct. I think the issue now occurs due to the input in the TOM. Where do TOM fetch the contents of the segmentations? I can't seem to find it.

**Comment by minar09** [comment]
It should be in cp_dataset.py file.

**Comment by RubReh** [comment]
Thanks! Checking and will get back to you. Really appreciate the quick answers! Also, the final results of TOM: Are they the ones in "p_rendered" or "try-on" ?

**Comment by minar09** [comment]
You are welcome. The "try-on" is the final result. We blend the "p_rendered" results with the warped_clothes to get the "try_on" results.

**Comment by RubReh** [comment]
@minar09 , I am getting closer and find that part of the reason for the issue in the TOM stage happens on line in the function

test_tom()

on line 156 in test.py:

`im_names = inputs['im_name']`

I find that if I sort this list of file names by:        

`im_names.sort()`

, I get better results where most of the try-ons look good. I found this through seeing that the composite image and the rendered image with the same file names sometimes camse from different images.

However, I am still seeing artifacts and mixed images that doesn't look right. I uploaded my entire infered output on the dropbox link below and would really appreciate if you could confirm that it's still in fact incorrect. Many outputs look good though as far as I can tell?

Do you have any idea what could cause these issues? I should also mentioned that I am running your code without Nvidia GPU and thus commented out all uses of cuda(). Do you think this can have any effect?

Thanks again for all your help!


Dropbox: https://www.dropbox.com/sh/db2d9tsv4e4038f/AAB2m6QvG8KflVTbumKNW5zWa?dl=0

Successful cases:

![006640_0](https://user-images.githubusercontent.com/38507268/89104885-dec75680-d41c-11ea-8606-f9ab909c5184.jpg)


Unsuccessful cases? 

![006145_0](https://user-images.githubusercontent.com/38507268/89104882-d4a55800-d41c-11ea-952c-5cddf869a27c.jpg)

![006976_0](https://user-images.githubusercontent.com/38507268/89104908-01f20600-d41d-11ea-8591-02e32420c764.jpg)

**Comment by minar09** [comment]
Hi @RubReh , good to know that you are getting better results. Yes, some of the results still are not as expected. For GMM and TOM results,  maybe we shouldn't sort the file names,  since the files' ordering comes from the test_pairs.txt file. If you can debug the inputs of the wrong ones, I think you can find the mismatches. CPU or GPU maybe doesn't have any effect on this. Sorry that I can't be of help much right now. Thank you very much for your efforts.

**Comment by RubReh** [comment]
Hi again @minar09. Seems the batch size is causing the issue. I seem to generate the correct dataset as long as I change the batch size to 1 instead of 4 as in your default. Can you confirm that this in fact looks like your results?

https://www.dropbox.com/sh/ooau1wl9d0od53y/AAAe4YBc48Mq41bA8AZ2h8Esa?dl=0

If you still need it I can create a PR for the data prep fix and add a readme line about the batch sizing?

**Comment by minar09** [comment]
Hi @RubReh , yeah, these results look like as the expected ones. Thank you very much for your time and effort.
Sure, your pull request is welcome. That would be a great help for others trying out our repository. Thanks.

**Comment by koushiksura** [comment]
Hi @RubReh , I am facing this issue, Could you please help me fix it. 
Thanks a lot in advance.
